src/plots/cartesian/axes.js

```diff
@@ -2146,6 +2146,46 @@ axes.doTicksSingle = function(gd, arg, skipTitle) {
         return trace.fill && trace.fill.charAt(trace.fill.length - 1) === axLetter;
     }
 
+    function lineNearZero(ax2, position) {
+        if(!ax2.showline || !ax2.linewidth) return false;
+        var tolerance = Math.max((ax2.linewidth + ax.zerolinewidth) / 2, 1);
+
+        function closeEnough(pos2) {
+            return typeof pos2 === 'number' && Math.abs(pos2 - position) < tolerance;
+        }
+
+        if(closeEnough(ax2._mainLinePosition) || closeEnough(ax2._mainMirrorPosition)) {
+            return true;
+        }
+        var linePositions = ax2._linepositions || {};
+        for(var k in linePositions) {
+            if(closeEnough(linePositions[k][0]) || closeEnough(linePositions[k][1])) {
+                return true;
+            }
+        }
+    }
+
+    function anyCounterAxLineAtZero(counterAxis, rng) {
+        var mainCounterAxis = counterAxis._mainAxis;
+        if(!mainCounterAxis) return;
+
+        var zeroPosition = ax._offset + (
+            ((Math.abs(rng[0]) < Math.abs(rng[1])) === (axLetter === 'x')) ?
+            0 : ax._length
+        );
+
+        var counterLetterAxes = axes.list(gd, counterLetter);
```

Comment:
It would nice to avoid looping over all axes here.

Can we skip this loop when `fullLayout._plots[counterAxis._mainSubplot].overlays` is empty, and only check `lineNearZero(counterAxis, zeroPosition)`?

Reply:
good call - it was almost that simple. one more relevant test in https://github.com/plotly/plotly.js/pull/2936/commits/bcd7c0666c80250dd4c78f230032df4facad2c37 and 🐎 in https://github.com/plotly/plotly.js/pull/2936/commits/0f59ca2b1bc0189960c1d558ba7414c1e3b54e2b

Reply:
Good to hear it was that easy. :100: